rewrite-java/src/test/java/org/openrewrite/java/recipes/ArraysAsListToImmutableRecipeTest.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.recipes;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class ArraysAsListToImmutableRecipeTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new ArraysAsListToImmutableRecipe());
+    }
+
+    @Test
+    void wrapArrayListToBeUnmodifiableWhenStrings() {
+        rewriteRun(
+          java("""
+              import java.util.Arrays;
+              
+              class A {
+                 public static final List<String> entries = Arrays.asList("A", "B");
```

Comment:
IIRC, `Arrays.asList` already returns a list that is unmodifiable.

Reply:
Ah, I'm wrong, you can't append to it, but you can replace existing values.

Reply:
Indeed; more details discussed in [this thread](https://github.com/openrewrite/rewrite/issues/2413#issuecomment-1586767731); I like your suggestion of using data flow to make this recipe safe to execute. I've not used data flow myself before, but would be good to explore here.

Reply:
To use data flow, this recipe would need to be moved to the `cleanup` package in `rewrite-static-analysis`

Reply:
That's indeed perhaps a better place for this recipe; sorry about that @marynasavchenko . Would you mind targeting https://github.com/openrewrite/rewrite-static-analysis/tree/main/src/main/java/org/openrewrite/staticanalysis for these files?
As seen above that's necessary to use data flow, which we need to be sure we can safely apply this recipe.